applications/GeoMechanicsApplication/custom_utilities/process_utilities.cpp

```diff
@@ -0,0 +1,44 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+
+// Project includes
+#include "process_utilities.h"
+
+namespace Kratos
+{
+std::vector<std::reference_wrapper<ModelPart>> ProcessUtilities::GetModelPartsFromSettings(
+    Model& rModel, const Parameters& rProcessSettings, const std::string& rProcessInfo)
+{
+    KRATOS_ERROR_IF_NOT(rProcessSettings.Has("model_part_name") ||
+                        rProcessSettings.Has("model_part_name_list"))
+        << "Please specify 'model_part_name' or 'model_part_name_list' for " << rProcessInfo;
+
+    KRATOS_ERROR_IF(rProcessSettings.Has("model_part_name") &&
+                    rProcessSettings.Has("model_part_name_list"))
+        << "The parameters 'model_part_name' and 'model_part_name_list' are mutually exclusive for "
+        << rProcessInfo;
+
+    const auto model_part_names = rProcessSettings.Has("model_part_name_list")
+                                      ? rProcessSettings["model_part_name_list"].GetStringArray()
+                                      : std::vector{rProcessSettings["model_part_name"].GetString()};
+    KRATOS_ERROR_IF(model_part_names.empty()) << "The parameters 'model_part_name_list' needs "
+                                                 "to contain at least one model part name for "
+                                              << rProcessInfo;
+
+    std::vector<std::reference_wrapper<ModelPart>> result;
```

Comment:
To make sure we allocate the memory that we need just once, I would set the capacity here:
```suggestion
    std::vector<std::reference_wrapper<ModelPart>> result;
    result.reserve(model_part_names.size());
```

Reply:
added